src/kafka-util/src/bin/kgen.rs

```diff
@@ -102,12 +125,26 @@ fn main() -> anyhow::Result<()> {
     let dd = Uniform::new_inclusive(0, 255);
     let mut buf = vec![];
     let mut rng = thread_rng();
+
+    let keys_random = matches.value_of("keys").unwrap() == "random";
+    let key_dist = if keys_random {
+        let key_min: u64 = matches.value_of("key-min").unwrap().parse()?;
+        let key_max: u64 = matches.value_of("key-max").unwrap().parse()?;
+        Some(Uniform::new_inclusive(key_min, key_max))
+    } else {
+        None
+    };
     for i in 0..n {
         if i % 10000 == 0 {
             eprintln!("Generating message {}", i);
         }
         gen_value(&mut buf, &ld, &dd, &mut rng);
-        let key = i.to_le_bytes();
+        let key_i = if let Some(key_dist) = key_dist.as_ref() {
+            key_dist.sample(&mut rng)
+        } else {
+            i as u64
+        };
+        let key = key_i.to_be_bytes();
         let mut rec = BaseRecord::to(topic)
```

Comment:
test_util::kafka::kafka_client abstracts a lot of the sending things to kafka / creating a new kafka topic logic away if you want to use that?

Reply:
OK, I'll check it out next time I need to make more changes.